backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java

```diff
@@ -24,21 +30,41 @@ public PhylogeneticNode() {
         nameLabel = "";
         distance = 0.;
         children = new ArrayList<>();
+        genomes= new ArrayList<>();
     }
 
     /**
-     * Initialize this node from tree node.
+     * Initialize this node from tree node, recursively.
      * @param node The node.
+     * @param parent The parentNode of this node, null if root.
      * @param distance The distance.
+     * @param id The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(TreeNode node, double distance) {
+    public PhylogeneticNode(TreeNode node, PhylogeneticNode parent, double distance, int id) {
```

Comment:
which is why I will write a controller class, as the field is needed in the backend. It's mathematically impossible to store the genomes at the right node if I can not iterate upwards through the tree.